### DISCUSSION TITLE
Smart phone charger control for Android with Tasker

### DISCUSSION THREAD

**Discussion by JoernBr** [arendst/Tasmota#14576] [Ideas]
Hi,
does somebody know how to control a phone charger by a NousA1 plug (using Tasmota 10.1.0) in that way, to turn off the Nous when the phone reached a charging level of, let´s say, 80% ? Right now, I´m doing it in a way using the Nous as timer, switch it on one hour before I get up in the morning (start charge) and switch off (stop charge) 5 min before my alarm makes me get up. But then the finally reached charging level is undefined. The idea is to stop the charge before the battery is 100% full.
I imagine to use an app on the (not rooted) Android phone like Tasker to read the battery level and then send a signal to the Nous to switch off the power supply to the charger. Is that possible?
This is probably a good way to extend battery life. Unfortunately I´m not able to program this.
Can someone help?

**Comment by ascillato** [comment]
Tasker can send HTTP commands, so you can make Tasker to send power off to your PLUG. More infor about HTTP commands at https://tasmota.github.io/docs/Commands/#with-web-requests

**Reply by barbudor** [reply]
The NousA1 has power monitoring.
So when the phone is charged power or current will fall under a given threshold and you switch the Nous off.
I'm using similar rules to switch off PC power when power consumption as fallen under a give  value for 30 seconds.

**Reply by sfromis** [reply]
Using energy monitoring would not work to cut off charging *before* the battery is full. 

As long as you have a decently functional app able to monitor charge level, it is "no problem" to send a command to Tasmota. You could even run Node-Red on the Android phone to do it.

**Reply by JoernBr** [reply]
> The NousA1 has power monitoring. So when the phone is charged power or current will fall under a given threshold and you switch the Nous off. I'm using similar rules to switch off PC power when power consumption as fallen under a give value for 30 seconds.

Yes, I found a rule like that. It works as well, but only shuts down the charger when the battery is full already.

**Reply by barbudor** [reply]
Adding a tool such as Tasker that will run in the background and sucks more battery has some clear drawbacks though.
Facebook, Discord, WhatsApp running in the background have a clear visible impact on battery.
It is generally admitted that the best battery management is to charge up to 100%, stop, and then do not recharge until the battery is really low in order to only get full cycles.

**Reply by sfromis** [reply]
If the "80%" was really a good strategy, I'd expect this to be a common feature for vendors to implement as a option. Only place I've seen something similar was on a laptop, where you can configure battery preservation using an "always on charger" mode to not go to 100% and stay there.

My new phone uses adaptive charging, slows down charging to hit full battery shortly before I wake up. Again with the goal of not staying at 100% while still on the charger.

**Reply by ascillato** [reply]
I have used tasker in the past and I couldn't notice more battery drain than usual. Seems that it wakes up only when there is something that triggers it. I had no issue with it. I'm not using anymore but in the past I linked my phone alarm to wake up with an automation in tasker that was sending an http command to a Tasmotized light. So whenever my phone's alarm goes off, after 5 minutes that anoying light turned on to force me to wake up.
So, depending on what type of automation you want, you can use Tasker without issues IMHO.
